model_api/cpp/utils/src/async_infer_queue.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+// Copyright (C) 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "utils/async_infer_queue.hpp"
+
+
+AsyncInferQueue::AsyncInferQueue(ov::CompiledModel& model, size_t jobs) {
+    if (jobs == 0) {
+        jobs = static_cast<size_t>(model.get_property(ov::optimal_number_of_infer_requests));
+    }
+
+    m_requests.reserve(jobs);
+    m_user_ids.reserve(jobs);
+
+    for (size_t handle = 0; handle < jobs; handle++) {
+        // Create new "empty" InferRequest without pre-defined callback and
+        // copy Inputs and Outputs from ov::CompiledModel
+        m_requests.emplace_back(model.create_infer_request());
+        m_user_ids.push_back(nullptr);
+        m_idle_handles.push(handle);
+    }
+
+    set_default_callbacks();
+}
+
+AsyncInferQueue::~AsyncInferQueue() {
+    m_requests.clear();
```

Comment:
An empty destructor would destroy `m_requests` anyway. I think you can keep `~AsyncInferQueue() = default;`

Reply:
Deleted